includes/background-jobs/class-sensei-scheduler-action-scheduler.php

```diff
@@ -0,0 +1,73 @@
+<?php
+/**
+ * This file contains Sensei_Scheduler_Action_Scheduler class.
+ *
+ * @package sensei
+ */
+
+if ( ! defined( 'ABSPATH' ) ) {
+	exit; // Exit if accessed directly.
+}
+
+/**
+ * Implementation of Sensei's scheduler using Action Scheduler.
+ *
+ * @since 3.0.0
+ */
+class Sensei_Scheduler_Action_Scheduler implements Sensei_Scheduler_Interface {
+	const ACTION_SCHEDULER_GROUP = 'sensei-enrolment';
+
+	/**
+	 * Schedule a job to run.
+	 *
+	 * @param Sensei_Background_Job_Interface $job  Job object.
+	 * @param int|null                        $time Time when the job should run. Defaults to now.
+	 *
+	 * @return mixed
+	 */
+	public function schedule_job( Sensei_Background_Job_Interface $job, $time = null ) {
+		if ( null === $time ) {
+			$time = time();
+		}
+
+		$next_scheduled_action = as_next_scheduled_action( $job->get_name(), [ $job->get_args() ], self::ACTION_SCHEDULER_GROUP );
+
+		if ( ! $next_scheduled_action ) {
+			as_schedule_recurring_action( $time, 0, $job->get_name(), [ $job->get_args() ], self::ACTION_SCHEDULER_GROUP );
```

Comment:
Fixed b500db1e2b69d37a86cce7a455ccfe14fcecff9c

Reply:
Here I mean some note why we are using the `as_schedule_recurring_action` and not the `as_schedule_single_action`. I think it's because some bug related to the `as_next_scheduled_action`, right?

But it's not critical, I'm doing some tests and I'll approve anyway if it's the only thing ;)